### PR TITLE
Bug 1209866 - Move tv_apps from mozL10n.DateTimeFormat to Intl API

### PR DESCRIPTION
**Pull request by zbraniecki**
https://bugzilla.mozilla.org/show_bug.cgi?id=1209866

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f8f235b4cb89efbc71a49dc4853dbcb8c7dee8b3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c1d39e42337abfe2636373909b9f35d463729e76

File: tv_apps/smart-system/js/value_selector/value_selector.js [hunk 1]
```diff
@@ -529,12 +531,8 @@
...
     }
 
-    var separator = ':';
-    var minutesPosition = localeTimeFormat.indexOf('%M');
-    if (minutesPosition > 0) {
-      separator = localeTimeFormat.substr(minutesPosition - 1, 1);
-    }
-    this.elements.hoursMinutesSeparator.textContent = separator;
+    this.elements.hoursMinutesSeparator.setAttribute('data-l10n-id',
+      'hourMinutesSeparator');
```
Comment (evelynhung) [inline tv_apps/smart-system/js/value_selector/value_selector.js]: I didn't find this l10n id being added, should we?

Comment (zbraniecki) [inline tv_apps/smart-system/js/value_selector/value_selector.js]: It's been added to `shared/locales/date/date.properties` - same file that l10n_date.js was using, so you should already have this file tracked.

File: tv_apps/smart-home/js/home.js [hunk 2]
```diff
@@ -779,19 +780,23 @@
...
-      this.timeElem.innerHTML = formatted;
-      this.timeElem.dataset.ampm = use12Hour ? f.localeFormat(now, '%p') : '';
+      this.timeElem.innerHTML = f.format(now);
+      if (use12Hour) {
+        var amPmFmt = mozIntl.DateTimeFormat(navigator.languages, {
+          hour12: true,
+          dayperiod: true,
+        });
+        this.timeElem.dataset.ampm = use12Hour ? amPmFmt.format(now) : '';
+      }
```
Comment (rexboy7) [inline tv_apps/smart-home/js/home.js]: I got `this.timeElem.innerHTML = '3:52 PM'` and `this.time.dataset.ampm = ''` on my build, which should be `3:52` and `PM` correspondingly. Not sure if it's a bug in `mozIntl`. If yes we can just commit it for now and fix it on follow-up.

Comment (rexboy7) [inline tv_apps/smart-home/js/home.js]: Seems it works on my phone; but not working on B2G-desktop of Oct. 6. Do you have any idea on this?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e72b2b04a72c8be507153c88a5b2553caa869798

File: tv_apps/smart-home/js/home.js (hunk 2 above)
Comment (zbraniecki) [inline tv_apps/smart-home/js/home.js]: I tested it on B2G-Desktop with different timezones (Shenghai, Warsaw, Los Angeles) and it worked properly. I rebased the patch on top of master. Can you retest pls?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4ca7d6f654f00272ac00db582e30b94ea1022a67

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d528b4f1292ff8a747be26dc11d4178e2e78177f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5c0b64eff0bc9a357bbe7193d69f15919f470056

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=511fd6e06814065cba13e9fc72e839b4298d254a

**Pull request merged**